lib/FtpConnection.js

```diff
@@ -1213,9 +1230,24 @@ FtpConnection.prototype._STOR_usingWriteFile = function(filename, flag) {
     time: startTime,
   });
 
-  self.respond('150 Ok to send data', function() {
-    self._whenDataReady(handleUpload);
-  });
+  // Filter extensions if whitelist available
+  var whitelist = self.server.options.allowedExtensions;
+  var accepted = true;
+  if (whitelist && typeof whitelist === 'object' && Array.isArray(whitelist) && whitelist.length) {
+    accepted = false;
+    whitelist.forEach(function (ext) {
+      if (filename.endsWith(ext)) accepted = true;
+    });
+  }
```

Comment:
Since we're repeating this whitelist checking logic, I think it would make sense to put it in a function so we can do: `var accepted = checkWhitelist(self.server.options.allowedExtensions, filename);`